HealthcareApp/src/Healthcare.Application/Implementations/AuthenticationService.cs

```diff
@@ -0,0 +1,70 @@
+using Application.Abstractions;
+using Application.Abstractions.Decorators;
+using Application.DTOs.Login;
+using Application.DTOs.Register;
+using Domain.Entities;
+using Microsoft.AspNetCore.Identity;
+using static Domain.Constants.UserRolesConstants;
+
+namespace Application.Implementations;
+
+public class AuthenticationService : IUserAuthenticationService
+{
+    private readonly IUserManagerDecorator<ApplicationUser> _userManager;
+    private readonly ISignInManagerDecorator<ApplicationUser> _signInManager;
+    private readonly IRoleManagerDecorator<IdentityRole> _roleManager;
+    private readonly ITokenGeneratorService _generatorService;
+
+
+    public AuthenticationService(IUserManagerDecorator<ApplicationUser> userManager,
+        ISignInManagerDecorator<ApplicationUser> signInManager,
+        IRoleManagerDecorator<IdentityRole> roleManager, ITokenGeneratorService generatorService)
+    {
+        _userManager = userManager;
+        _signInManager = signInManager;
+        _roleManager = roleManager;
+        _generatorService = generatorService;
+    }
+
+    public async Task<IdentityResult> RegisterAsync(RegisterUserDTO registerUserDto)
+    {
+        if (!IsRoleAllowed(registerUserDto.Role))
+        {
+            string description = $"Invalid role. Allowed roles are: ";
+            description += string.Join(' ', AllowedRoles);
```

Comment:
Use string interpolation in such cases - $"Invalid role. {string.Join(' ', AllowedRoles)}"